packages/graphql/src/classes/Executor.ts

```diff
@@ -0,0 +1,215 @@
+
+            if (isSessionLike(this.executionContext)) {
+                return await this.sessionRun(query, parameters, defaultAccessMode, this.executionContext);
+            }
+
+            return await this.transactionRun(query, parameters, this.executionContext);
+        } catch (error) {
+            throw this.formatError(error);
+        }
+    }
+
+    private formatError(error: unknown) {
+        if (error instanceof Neo4jError) {
+            if (error.message.includes(`Caused by: java.lang.RuntimeException: ${AUTH_FORBIDDEN_ERROR}`)) {
+                return new Neo4jGraphQLForbiddenError("Forbidden");
+            }
+
+            if (error.message.includes(`Caused by: java.lang.RuntimeException: ${AUTH_UNAUTHENTICATED_ERROR}`)) {
+                return new Neo4jGraphQLAuthenticationError("Unauthenticated");
+            }
+
+            if (error.message.includes(`Caused by: java.lang.RuntimeException: ${RELATIONSHIP_REQUIREMENT_PREFIX}`)) {
+                const [, message] = error.message.split(RELATIONSHIP_REQUIREMENT_PREFIX);
+                return new Neo4jGraphQLRelationshipValidationError(message);
+            }
+
+            if (error.code === "Neo.ClientError.Schema.ConstraintValidationFailed") {
+                return new Neo4jGraphQLConstraintValidationError("Constraint validation failed");
+            }
+        }
+
+        debug("%s", error);
+
+        return error;
+    }
+
+    private generateQuery(query: string): string {
+        if (this.queryOptions && Object.keys(this.queryOptions).length) {
+            const queryOptions = `CYPHER ${Object.entries(this.queryOptions)
+                .map(([key, value]) => `${key}=${value}`)
+                .join(" ")}`;
+
+            return `${queryOptions}\n${query}`;
+        }
+
+        return query;
+    }
+
+    private generateParameters(query: string, parameters: any): Record<string, any> {
+        if (query.includes("$auth.") || query.includes("auth: $auth") || query.includes("auth:$auth")) {
+            return { ...parameters, auth: this.auth };
+        }
+
+        return parameters;
+    }
+
+    private getSessionParam(defaultAccessMode: SessionMode): SessionParam {
+        const sessionParam: SessionParam = { defaultAccessMode };
+
+        if (this.database) {
+            sessionParam.database = this.database;
+        }
+
+        if (this.bookmarks) {
+            sessionParam.bookmarks = this.bookmarks;
+        }
+
+        return sessionParam;
+    }
+
+    private getTransactionConfig(): TransactionConfig {
+        const app = `${environment.NPM_PACKAGE_NAME}@${environment.NPM_PACKAGE_VERSION}`;
+
+        return {
+            metadata: {
+                app,
+                type: "user-transpiled",
+            },
+        };
+    }
+
+    private async sessionRun(query: string, parameters, defaultAccessMode, session): Promise<QueryResult> {
+        const result = await session[`${defaultAccessMode.toLowerCase()}Transaction`](
```

Comment:
**nitpick, non-blocking** for readability purposes, maybe extract `${defaultAccessMode.toLowerCase()}Transaction` into a separate varaible "transactionName" or something like that

Reply:
Good shout!